### PR TITLE
add script to check core issue 2116

### PR DESCRIPTION
**Pull request by iCollin**
ATF Test Scripts to check https://github.com/smartdevicelink/sdl_core/issues/2116

This PR is **ready** for review.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/6_2/2116_OnButtonPress_LimitedHMILevel.lua [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+local appParams = {
+  [1] = { appName = "Test Application", appID = "0001",  fullAppID = "0000001" }
+}
+
+local function receiveButtonPressWithAppId()
+  local mobSession1 = common.mobile.getSession()
+
+  common.hmi.getConnection():SendNotification("Buttons.OnButtonEvent", {name = "OK", mode = "BUTTONDOWN", appID = common.app.getHMIId(1) })
+  mobSession1:ExpectNotification("OnButtonEvent",{buttonName = "OK", buttonEventMode="BUTTONDOWN"}):Times(1)
```
Comment (ShobhitAd) [inline test_scripts/Defects/6_2/2116_OnButtonPress_LimitedHMILevel.lua:34]:
```suggestion
  mobSession1:ExpectNotification("OnButtonEvent",{buttonName = "OK", buttonEventMode="BUTTONDOWN"})
```
`Times(1)` isn't necessary. The number of expectations defaults to 1 if `Times` is not specified.

File: test_scripts/Defects/6_2/2116_OnButtonPress_LimitedHMILevel.lua [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  common.hmi.getConnection():SendNotification("Buttons.OnButtonPress", {name = "OK", mode = "LONG", appID = common.app.getHMIId(1)})
+  mobSession1:ExpectNotification("OnButtonPress",{buttonName = "OK", buttonPressMode = "LONG"}):Times(1)
```
Comment (ShobhitAd) [inline test_scripts/Defects/6_2/2116_OnButtonPress_LimitedHMILevel.lua:36]:
```suggestion
  mobSession1:ExpectNotification("OnButtonPress",{buttonName = "OK", buttonPressMode = "LONG"})
```
`Times(1)` isn't necessary. The number of expectations defaults to 1 if `Times` is not specified.

File: test_scripts/Defects/6_2/2116_OnButtonPress_LimitedHMILevel.lua [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    mobSession1:ExpectNotification("OnButtonEvent",{buttonName = "OK", buttonEventMode="BUTTONDOWN"}):Times(0)
+    common.hmi.getConnection():SendNotification("Buttons.OnButtonPress", {name = "OK", mode = "LONG"})
+    mobSession1:ExpectNotification("OnButtonPress",{buttonName = "OK", buttonPressMode = "LONG"}):Times(0)
+  end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL and HMI", common.start)
+runner.Step("Connect two mobile devices to SDL", common.connectMobDevices, {devices})
```
Comment (ShobhitAd) [inline test_scripts/Defects/6_2/2116_OnButtonPress_LimitedHMILevel.lua:52]:
```suggestion
runner.Step("Connect mobile device to SDL", common.connectMobDevices, {devices})
```

File: test_scripts/Defects/6_2/2116_OnButtonPress_LimitedHMILevel.lua [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+runner.Step("Register App1 from device 1", common.registerAppEx, { 1, appParams[1], 1 })
```
Comment (ShobhitAd) [inline test_scripts/Defects/6_2/2116_OnButtonPress_LimitedHMILevel.lua:53]:
```suggestion
runner.Step("Register App1", common.registerAppEx, { 1, appParams[1], 1 })
```
No need to specify the device number if there's only one device

**Comment by iCollin** [comment]
@ShobhitAd Thank you for the good suggestions, they have been applied in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2410/commits/76b6da2245c10f41dc5f5fd6944ecbe6895c8b57

**Pull request merged**